### PR TITLE
[DEV-11495] Feature - Implement "Placeholders" and dedicated "Menu Item" for X (Twitter)

### PR DESCRIPTION
**Pull request by MohammadxAli**
# Menu Item

image

# Placeholders

image
image

### CODE REVIEW

**Comment by MohammadxAli** [comment]
Since options are too many, I will merge each 3-4 options into a single PR from now on to avoid noise and make reviewing easier.

cc @e1himself @alpower

File: packages/slate-editor/src/extensions/placeholders/elements/SocialPostPlaceholderElement.tsx [hunk 1]
```diff
@@ -136,6 +136,17 @@ export function SocialPostPlaceholderElement({
         });
     }
 
+    if (element.provider === PlaceholderNode.Provider.X) {
+        return render({
+            inputTitle: 'X post',
+            inputDescription: 'Paste a social media link and hit Enter',
+            inputPlaceholder: 'https://www.x.com/post/',
```
Comment (e1himself) [inline packages/slate-editor/src/extensions/placeholders/elements/SocialPostPlaceholderElement.tsx:143]: I believe they haven't yet migrated the URLs to the new domain. So I suggest to keep it as `twitter.com/post/` to not confuse people. 

@alpower agree?

**Review by e1himself — APPROVED** [review]

File: packages/slate-editor/src/extensions/placeholders/elements/SocialPostPlaceholderElement.tsx (hunk 1 above)
Comment (MohammadxAli) [reply packages/slate-editor/src/extensions/placeholders/elements/SocialPostPlaceholderElement.tsx]: Good call 👍

Used "twitter" instead: https://github.com/prezly/slate/pull/475/commits/0f7a3c9db2ed7c58577edec64e301eb1293636bf

**Pull request merged**